### ISSUE TITLE
RFE: please provide support for upcoming LLVM 17.x

### ISSUE THREAD

**Issue by kloczek** [iovisor/bpftrace#2762]
At the moment it is 17.0.0rc4 and shortly should be available final 17.0.0.
It would be good to adjust bpftrace code to prepare that code for LLV 17.0.0.
```console
CMake Error at CMakeLists.txt:184 (message):
  Unsupported LLVM version found via /usr/include: 17


CMake Error at CMakeLists.txt:185 (message):
  Only versions between 6 and 16 are supported


CMake Error at CMakeLists.txt:186 (message):
  Specify an LLVM major version using LLVM_REQUESTED_VERSION=<major version>


-- Found LLVM 17.0.0: /usr/lib64/cmake/llvm
-- Could NOT find FFI (missing: FFI_LIBRARIES HAVE_FFI_CALL)
-- Could NOT find LibEdit (missing: LibEdit_INCLUDE_DIRS LibEdit_LIBRARIES)
-- Could NOT find LibXml2 (missing: LIBXML2_LIBRARY LIBXML2_INCLUDE_DIR)
-- Disabled codegen test for LLVM != 12
-- Found GTest: /usr/lib64/cmake/GTest/GTestConfig.cmake (found version "1.14.0")
-- Found GMock: /usr/lib64/libgmock.so
-- Found Threads: TRUE
-- Not building with LLVM 12, skipping codegen-validator test
-- Configuring incomplete, errors occurred!
```

**Comment by viktormalik** [comment]
We have #2667 which I just updated to contain everything necessary to build bpftrace with LLVM 17.

Unfortunately, LLVM 17 is not released in [NixOS/nixpkgs](https://github.com/NixOS/nixpkgs), yet, so we cannot run tests in the CI. Without proper testing, we're reluctant to support LLVM 17.

So, I'd suggest that until LLVM 17 is out, anyone who wants to build with LLVM 17 should pull #2667.

**Comment by kloczek** [comment]
OK 👍

**Comment by viktormalik** [comment]
Closed by #2667.
